Source/JavaScriptCore/runtime/JSObject.cpp

```diff
@@ -572,6 +572,8 @@ bool JSObject::getOwnPropertySlotByIndex(JSObject* thisObject, JSGlobalObject* g
     
     if (i > MAX_ARRAY_INDEX)
         return thisObject->methodTable()->getOwnPropertySlot(thisObject, globalObject, Identifier::from(vm, i), slot);
+
+    unsigned attributes = thisObject->structure()->typeInfo().arrayStorageVectorPropertiesAreReadOnly() ? static_cast<unsigned>(PropertyAttribute::ReadOnly) : static_cast<unsigned>(PropertyAttribute::None);
```

Comment:
Can you avoid `thisObject->structure()->typeInfo().arrayStorageVectorPropertiesAreReadOnly()` for most of indexing types? We should not load structure if it is not necessary. And I think most of indexing type never gets `arrayStorageVectorPropertiesAreReadOnly` = true. (Only slow-put ones)

Reply:
Yeah, probably we can just add a helper onto JSObject that does the fast thing depending on indexing type

Reply:
I've made the structure's `typeInfo()` check conditional based on indexing type, only for slow-puts. I'm not sure that at this point (only one usage) it's worth to extract a helper though.